engines/pytorch/pytorch-engine/src/main/java/ai/djl/pytorch/engine/PtNDArray.java

```diff
@@ -223,6 +230,369 @@ public NDArray stopGradient() {
+            ret[i] = bb.get() != 0;
+        }
+        return ret;
+    }
+
+    private ByteBuffer toDirectByteBuffer() {
+        // Check if the tensor's data type is STRING, if so, throw an exception
+        if (getDataType() == DataType.STRING) {
+            throw new UnsupportedOperationException(
+                    "toDirectByteBuffer is not supported for String tensor.");
+        }
+        // Check if the tensor is sparse, if so, throw an exception
+        if (isSparse()) {
+            throw new UnsupportedOperationException(
+                    "toDirectByteBuffer is not supported for sparse tensors.");
+        }
+        // Check if the tensor is not contiguous, if so, throw an exception
+        if (!isContiguous()) {
+            throw new UnsupportedOperationException(
+                    "toDirectByteBuffer requires the tensor to be contiguous.");
+        }
+        // Obtain a direct ByteBuffer from the tensor, set it to read-only, and ensure it uses the
+        // native byte order
+        return JniUtils.getDirectByteBuffer(this).asReadOnlyBuffer().order(ByteOrder.nativeOrder());
+    }
+
+    /**
+     * Checks if the current {@code NDArray} is contiguous.
+     *
+     * <p>If the data type of the {@code NDArray} is a string ({@code DataType.STRING}), this
+     * operation is not supported, and an {@code UnsupportedOperationException} exception is thrown.
+     *
+     * <p>For non-string types of {@code NDArray}, it calls the {@code JniUtils.isContiguous} method
+     * to determine if it is contiguous. A contiguous {@code NDArray} is stored continuously in
+     * memory, which may have performance advantages for certain operations.
+     *
+     * @return {@code true} if the {@code NDArray} is contiguous; otherwise, {@code false}.
+     * @throws UnsupportedOperationException if the data type of the {@code NDArray} is a string, as
+     *     this operation is not supported.
+     */
+    public boolean isContiguous() {
+        if (getDataType() == DataType.STRING) {
+            throw new UnsupportedOperationException(
+                    "isContiguous is not supported for String tensor.");
+        }
+
+        return JniUtils.isContiguous(this);
+    }
+
+    /**
+     * Converts the current {@code NDArray} to a contiguous format.
+     *
+     * <p>This method checks if the {@code NDArray} is already contiguous, not sparse, and not
+     * having a layout of 2. If these conditions are met, it simply duplicates the current {@code
+     * NDArray} to avoid unnecessary operations. Otherwise, it utilizes {@code
+     * JniUtils.toContiguous} to convert the {@code NDArray} into a contiguous format.
+     *
+     * <p>A contiguous {@code NDArray} is stored continuously in memory, which may have performance
+     * advantages for certain operations.
+     *
+     * @return A new {@code PtNDArray} that is guaranteed to be contiguous.
+     */
+    public PtNDArray toContiguous() {
```

Comment:
I don't feel we need to expose isContiguous() and toContiguous() in Java level. We should be able to handle this at `torchGetDirectByteBuffer()` level.